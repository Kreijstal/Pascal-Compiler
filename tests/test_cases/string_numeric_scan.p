program string_numeric_scan;
var
  Input: string;
  Position, Start: Integer;
  NumStr: string;
begin
  Input := '123.45';
  Position := 1;
  Start := Position;

  while (Position <= Length(Input)) and (Position in [1..6]) do
    Inc(Position);

  NumStr := '';
  while Start < Position do
  begin
    NumStr := NumStr + Input[Start];
    Inc(Start);
  end;

  WriteLn('Number string: ', NumStr);
end.
