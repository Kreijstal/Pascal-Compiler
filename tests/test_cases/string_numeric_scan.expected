Number string: 123.45
